loqusdbapi/utils.py

```diff
@@ -0,0 +1,280 @@
+        raise VCFParserError(f"Variant types in {vcf_path}: {snv_vcf_var_types}, required: snv")
+
+    case_object.nr_variants = snv_vcf_variant_count
+    case_object.vcf_path = vcf_path
+
+    return case_object
+
+
+def parse_sv_vcf(vcf_path: Union[Path, str], case_object: Case) -> Case:
+    sv_vcf = VCF(vcf_path, threads=settings.cyvcf_threads)
+    if not sv_vcf.contains("GQ"):
+        raise VCFParserError(f"GQ not found in {vcf_path}")
+    sv_vcf_variant_count = 0
+    sv_vcf_individuals = sv_vcf.samples
+    sv_vcf_var_types = set()
+
+    for variant in sv_vcf:
+        sv_vcf_variant_count += 1
+        sv_vcf_var_types.add(variant.var_type)
+
+    case_object.nr_sv_variants = sv_vcf_variant_count
+    case_object.vcf_sv_path = vcf_path
+    return case_object
+
+
+def parse_profiles(adapter: MongoAdapter, case_object: Case) -> Case:
+
+    profiles: dict = get_profiles(adapter=adapter, vcf_file=case_object.profile_path)
+    profile_vcf = VCF(case_object.profile_path, threads=settings.cyvcf_threads)
+    samples: List[str] = profile_vcf.samples
+    for sample_index, sample in enumerate(samples):
+        individual = Individual(
+            ind_id=sample,
+            case_id=case_object.case_id,
+            ind_index=sample_index,
+            profile=profiles[sample],
+        )
+        if case_object.vcf_path:
+            case_object.inds[sample] = individual
+            case_object.individuals.append(Individual(**individual.dict()))
+        if case_object.vcf_sv_path:
+            case_object.sv_inds[sample] = individual
+            case_object.sv_individuals.append(Individual(**individual.dict()))
+    return case_object
+
+
+def check_profile_duplicates(adapter: MongoAdapter, case_object: Case) -> Case:
+    for existing_case in adapter.cases():
+
+        if existing_case.get("individuals") is None:
+            continue
+
+        for individual in existing_case["individuals"]:
+            if not individual.get("profile"):
+                continue
+
+            for sample in case_object.individuals:
+                similarity = compare_profiles(sample.profile, individual["profile"])
+                if similarity >= settings.load_hard_threshold:
+                    raise ProfileDuplicationError(
+                        f"Profile of sample {sample.ind_id} "
+                        f"matches existing profile {individual.get('ind_id')}"
+                    )
+
+                elif similarity >= settings.load_soft_threshold:
+                    match = f"{existing_case['case_id']}.{individual['ind_id']}"
+                    sample.similar_samples.append(match)
+
+    return case_object
+
+
+def build_case_object(
+    adapter: MongoAdapter,
+    case_id: str,
+    profile_path: Union[Path, str],
+    vcf_path: Union[Path, str],
+    vcf_sv_path: Union[Path, str] = None,
+) -> dict:
+
+    # Create case object prior to parsing VCF files
+    case_object: Case = Case(
```

Comment:
maybe this case object could be another class that the Case object inherits from. Easier to talk about this. When you have time